Update Compiz ebuilds

ethus3h opened the pull request:
This PR updates the Compiz ebuilds to version 0.8.14.

Some discussions are at https://bugs.gentoo.org/532580 and https://github.com/compiz-reloaded/compiz/issues/86.

jonasstein commented on x11-misc/compiz-debug-utils/metadata.xml:
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>elliot@futuramerlin.com</email>
```
a developer and/or proxy-maint@gentoo.org is missing. Please discuss with the proxy team.

jonasstein commented on x11-misc/compiz-debug-utils/compiz-debug-utils-0_pre20171023.ebuild:
```diff
@@ -0,0 +1,28 @@
+
+myCommit="f700dd2b6af1f0c93b6b1eb1d75e658783f848e2"
+
+DESCRIPTION=" Helper utilities for debugging Compiz"
```
+DESCRIPTION="Helper utilities for debugging Compiz"

mgorny replied:
I think @jonasstein means to say you've got extra space here.

ethus3h commented:
@jonasstein: Ok! Added the proxy-maintainer info, also the <upstream>s to the metadata.xmls. Also fixed the extra space, and removed an extra comma I found in a <longdescription>. :)

monsieurp commented:
Hi @ethus3h. I saw your comment in the IRC channel and thought I would respond. Please provide build logs for those ebuilds as well as a screenshot with said version of compiz working. We can't test them all as it would be very much time consuming. Thank you for your understanding and bear with us.

ethus3h commented:
Build logs for new packages are attached to the bug reports for the corresponding new packages (bug number for each is in the commit message where that package was added). I don't have logs for the ones that are version bumps of old packages, didn't know they were needed. I'll add them in a comment here in a bit. Thanks!

Screenshot:
![screenshot_2017 10 27_20 36 55](https://user-images.githubusercontent.com/10457522/32129545-a9ed0eaa-bb56-11e7-9c4f-4dc9b64b34d2.png)

ethus3h commented:
Here's a build log for all the packages: https://gist.github.com/ethus3h/556dfadcd8f1caf8de9b1e0cf09f5b3e

matijaskala commented on x11-apps/fusion-icon/fusion-icon-0.2.4-r1.ebuild:
```diff
@@ -0,0 +1,66 @@
+		dev-libs/libappindicator:3
+		x11-wm/compiz[gtk3]
+	)
+	!gtk3? ( x11-wm/compiz[gtk(+),-gtk3] )
```
Is this actually needed? As far as I know compiz uses gtk only for decorations.

matijaskala replied:
One can install `fusion-icon` with `gtk2` alongside `compiz` with `gtk3` or vice versa. They are using gtk independently from each other so there shouldn't be any issues. You can drop lines `29`, `33` and `35`.

ethus3h commented:
Sorry for the delay. I've made the requested dependency change, and changed the bug report references to GLEP 66 format. I'm not really sure how to fix the Manifest file conflicts that are now showing up. What's next? Thanks! :)

a17r commented:
>I'm not really sure how to fix the Manifest file conflicts that are now showing up.

Just remanifest so they get a proper BLAKE2B checksum.

ethus3h commented:
Um, I rebased and went through them all and ran `repoman manifest` with repoman-2.3.3, but it still gives SHA256/SHA512/WHIRLPOOL for all of them, no BLAKE2B. I don't see anything about BLAKE2B in `man repoman`, [the Gentoo Wiki page](https://wiki.gentoo.org/wiki/Repoman), or Google searching for "blake2b repoman"... How to do this? Thanks! :)

ethus3h commented:
(I'll also add `bugs-to` in metadata.xml which I should do, and update the year in the headers, when doing the new checksums.)

ethus3h commented:
Finally figured it out regarding the BLAKE2B! — had to update the metadata/layout.conf to get BLAKE2B to be generated. Will rebase+push shortly. Thanks!

mgorny commented on x11-wm/compiz/Manifest:
```diff
@@ -0,0 +1 @@
+DIST compiz-0.8.14.tar.xz 1137696 BLAKE2B 7a353bba661f5d81092f6850448524c44cdb2fbab1c65db994bf78dc193f131874c83664b64c461de9f4a63575af710088fdb2c0adcc3c6d0d4e06949c618ccc SHA512 ce02c7ac0ba864a68ace05abf4b9cb833185cf47a2dcaf5aa9e900336a4ecff5107ef503646bdd6ec675d9fa7a54bb5ae07df4f1cadea098765e5a5b4ca80d5e
```
Don't call it update, it's a new package at this moment. Calling it 'update' makes me wth are older ebuilds.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+LICENSE="GPL-2+ LGPL-2.1 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cairo compizconfig dbus fuse gsettings +gtk gtk3 inotify +svg marco mate"
```
Please sort IUSE lexically by flag name (ignoring the `+`).

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+}
+
+src_configure() {
+	local myconf=""
```
Please make `myconf` into an array, and put all arguments you'd otherwise pass to `econf` in it.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+src_configure() {
+	local myconf=""
+	if use gtk;
+		then myconf=" --with-gtk=$(usex gtk3 3.0 2.0)"
```
Put `then` on the same line as `if`, or in a separate line but not with the following commands.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+		myconf=" --disable-gtk"
+	fi
+
+	econf \
```
... and then you'd have `econf "${myconf[@]}"`

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+
+src_install() {
+	default
+	prune_libtool_files --all
```
This is deprecated, please replace with:

    find "${D}" -name '*.la' -delete || die

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+}
+
+pkg_preinst() {
+	use gsettings && gnome2_schemas_savelist
```
This is no longer necessary.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+
+EAPI=6
+
+inherit autotools eutils gnome2-utils
```
You're not using `eutils` anywhere. You were using `ltprune` but you're asked to remove it.

mgorny commented on x11-wm/compiz/metadata.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<use>
```
Pleas put `<use/>` below maintainers; when we open this file we usually need to read maintainers, so it's best to keep them on top.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,119 @@
+	x11-libs/pango
+	cairo? ( >=x11-libs/cairo-1.4[X] )
+	compizconfig? (
+		>=x11-libs/libcompizconfig-0.8
```
You need to order the commits so that the dependencies come first, i.e. that every commit is green by `repoman full -d` (and yes, you're supposed to test that).

mgorny commented:
Ping. Are you still there?

ethus3h commented:
Hi, yes, just been busy with university.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,111 @@
+
+DEPEND="${COMMONDEPEND}
+	virtual/pkgconfig
+	x11-proto/damageproto
```
Today this should use the combined package:

    x11-base/xorg-proto

I'm sorry but it's been added since we last reviewed.

mgorny commented on x11-wm/compiz/compiz-0.8.14.ebuild:
```diff
@@ -0,0 +1,111 @@
+	fi
+
+	myconf+=(
+		--enable-fast-install \
```
You don't need backslashes inside array.

mgorny commented on x11-wm/compiz/metadata.xml:
```diff
@@ -0,0 +1,26 @@
+	<maintainer type="person">
+		<email>elliot@futuramerlin.com</email>
+		<name>Elliot Chandler</name>
+		<description>Proxied maintainer</description>
```
Please remove the obvious `<description/>`s.

mgorny commented on x11-libs/libcompizconfig/libcompizconfig-0.8.14.ebuild:
```diff
@@ -0,0 +1,37 @@
+DEPEND="${RDEPEND}
+	>=dev-util/intltool-0.41
+	virtual/pkgconfig
+	x11-proto/xproto
```
Likewise.

mgorny commented on x11-libs/libcompizconfig/libcompizconfig-0.8.14.ebuild:
```diff
@@ -0,0 +1,37 @@
+	x11-proto/xproto
+"
+
+RESTRICT="test"
```
Please add a comment above explaining why you needed to restrict tests.

mgorny commented on x11-plugins/compiz-plugins-main/compiz-plugins-main-0.8.14.ebuild:
```diff
@@ -0,0 +1,42 @@
+
+src_install() {
+	default
+	prune_libtool_files
```
Please try removing them and see if things still work ;-).

mgorny commented on x11-plugins/compiz-plugins-extra/compiz-plugins-extra-0.8.14.ebuild:
```diff
@@ -0,0 +1,50 @@
+
+RDEPEND="
+	gnome-base/librsvg
+	libnotify? ( x11-libs/libnotify )
```
Please move USE-conditional deps below unconditional deps. This is a standard practice to improve readability.

mgorny commented on x11-plugins/compiz-plugins-community/compiz-plugins-community-0.8.14.ebuild:
```diff
@@ -0,0 +1,60 @@
+
+EAPI=6
+
+if [[ "${PV}" == "9999" ]]; then
```
Why are you changing ebuild style in the middle of the series? It looks weird-ish.

mgorny commented on x11-plugins/compicc/compicc-0.8.10.ebuild:
```diff
@@ -0,0 +1,32 @@
+
+EAPI=6
+
+inherit eutils
```
You don't seem to be using eutils here.

mgorny commented on x11-plugins/compiz-extra-snowflake-textures/compiz-extra-snowflake-textures-0.8.14.ebuild:
```diff
@@ -0,0 +1,29 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Shouldn't this come before -meta?

mgorny commented on x11-plugins/compiz-extra-snowflake-textures/compiz-extra-snowflake-textures-0.8.14.ebuild:
```diff
@@ -0,0 +1,29 @@
+
+src_install() {
+	insinto /usr/share/compiz/snow/
+	GLOBIGNORE="README.md:.git:.egup.tags"
```
But `*` doesn't match dotfiles, does it? That said, I'd rather you not use magic of `GLOBIGNORE` and `rm README.md` instead.

ethus3h commented:
Hi again @mgorny, I think this finishes the most recent batch of requested changes. Thanks!

mgorny commented on dev-python/compizconfig-python/compizconfig-python-0.8.14.ebuild:
```diff
@@ -0,0 +1,45 @@
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+src_configure() {
+	local myeconfargs=(
```
`default` does not use `myeconfargs`.

mgorny commented on x11-misc/ccsm/ccsm-0.8.14.ebuild:
```diff
@@ -0,0 +1,57 @@
+	# return error if wrong arguments passed to setup.py
+	sed -i -e 's/raise SystemExit/\0(1)/' setup.py || die 'sed on setup.py failed'
+
+	# correct gettext behavior
```
Please remove this block. It's not strictly required, and really belong upstream.

mgorny commented on x11-misc/simple-ccsm/simple-ccsm-0.8.14.ebuild:
```diff
@@ -0,0 +1,57 @@
+}
+
+pkg_preinst() {
+	gnome2_icon_savelist
```
savelist is no longer used.

mgorny commented on x11-misc/ccsm/ccsm-0.8.14.ebuild:
```diff
@@ -0,0 +1,57 @@
+
+python_configure_all() {
+	mydistutilsargs=(
+		build \
```
Backslashes are unnecessary inside array.

mgorny commented on x11-misc/compiz-debug-utils/compiz-debug-utils-0_pre20171023.ebuild:
```diff
@@ -0,0 +1,29 @@
+
+EAPI=6
+
+myCommit="f700dd2b6af1f0c93b6b1eb1d75e658783f848e2"
```
Global variables should be uppercase, please.

mgorny commented on x11-misc/compiz-debug-utils/compiz-debug-utils-0_pre20171023.ebuild:
```diff
@@ -0,0 +1,29 @@
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+		>=dev-lang/python-3:*
```
Using Python is allowed only via python*r1 eclasses. I think python-single-r1 would be a good fit here. Also, too much indent.

mgorny commented on x11-misc/compiz-debug-utils/compiz-debug-utils-0_pre20171023.ebuild:
```diff
@@ -0,0 +1,29 @@
+src_install() {
+	dodoc README.md
+	rm README.md
+	dobin *
```
This installs `/usr/bin/COPYING`

mgorny commented on x11-apps/compiz-manager/compiz-manager-0.7.0.ebuild:
```diff
@@ -0,0 +1,22 @@
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="x11-apps/mesa-progs
+		x11-apps/xdpyinfo
```
Too much indent.

mgorny commented on x11-apps/compiz-boxmenu/compiz-boxmenu-1.1.12.ebuild:
```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2018 Gentoo Foundation
```
This one is completely missing python eclass use.

mgorny commented on x11-apps/fusion-icon/fusion-icon-0.2.4-r1.ebuild:
```diff
@@ -0,0 +1,64 @@
+python_configure_all() {
+	mydistutilsargs=(
+		build \
+		--with-gtk=$(usex gtk3 3.0 2.0)
```
How is this going to work when only qt is enabled?

mgorny commented on x11-apps/fusion-icon/fusion-icon-0.2.4-r1.ebuild:
```diff
@@ -0,0 +1,64 @@
+python_install_all() {
+	mydistutilsargs=(
+		install \
+		--prefix=/usr/local
```
What are you trying to achieve here? Installing to `/usr/local` is forbidden.

matijaskala commented on x11-plugins/compiz-plugins-main/compiz-plugins-main-0.8.14.ebuild:
```diff
@@ -0,0 +1,42 @@
+	<x11-libs/compiz-bcop-0.9
+	>=x11-wm/compiz-0.8
+	<x11-wm/compiz-0.9
+	x11-libs/cairo
```
According to the configure scripts, `cairo-plugins-main` and `cairo-plugins-experimental` require `x11-libs/cairo[X]`.

matijaskala commented on x11-plugins/compiz-plugins-experimental/compiz-plugins-experimental-0.8.14.ebuild:
```diff
@@ -0,0 +1,48 @@
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	gnome-base/librsvg
```
The following dependencies are missing here:
- media-libs/glew:=
- net-misc/curl
- x11-libs/cairo[X]
- x11-libs/libXScrnSaver
- x11-plugins/compiz-plugins-extra

ethus3h commented on x11-wm/emerald/metadata.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<use>
```
oops, I should have caught that...

mgorny commented:
> @mgorny, you mention emerald segfaults. I haven't had any problem with it, using the live ebuild in what is mostly stable amd64 for about a year. How do you test on a plain system to avoid configuration-specific differences? Using the Live DVD? In regards to this specific problem, perhaps a backtrace would help, since I can't reproduce it. Sorry.

I suspect it's related to the fact that I'm using open-source Radeon drivers. I recall that back in the day those packages were really picky about hardware. Don't worry about it; we can merge it as-is and see if somebody else can reproduce the problem.

> 
> Also, how do you get the dependency.equalsversion report? I don't see that QA keyword in man repoman, and can't find it in a Web search. Similarly, I don't get the reports mentioned in this comment. I assume there's some configuration option for extra QA notices I have not set, but I don't know what they are...

I suppose it was never merged into mainline Portage. It's in `sys-apps/portage-mgorny` if you want to try it.

> Where you mentioned incorrect dependency lists — is there a script or something for checking what dependencies something uses, or are you able to figure it out from the packages' source code (the configure.ac perhaps, for these packages)? It would be nice if there's something where I can build and run the app, and then it will spit out a list of all the packages that it used files from (like equery belongs does)... I haven't learned Autotools yet, it makes my head spin -_-;

`configure.ac` + `readelf -d` on installed executables.

nkichukov commented:
Hi, any progress with this? Compiz-reloaded is active and maintained and works smooth without any issues for me on amd64 arch.

ethus3h commented:
@nkichukov The pull request was coming along, but then Gentoo's GitHub account got compromised (see https://www.gentoo.org/news/2018/06/28/Github-gentoo-org-hacked.html) and they vandalized the pull request.
I don't know how to fix it, and it would take a lot of time to rewrite it from scratch.

mgorny commented:
Don't you have a local copy of the branch? You should be able to just `git push -f` it.

nkichukov commented:
Yeah, I discovered your overlay when I was working on switching from compiz 0.9 (canonical) back to 0.8 (reloaded), so I have been cherry-picking from there(great work btw!). Everything looks good and works just fine. Now that they were approved for inclusion with all the remarks and corrections it should be as simple as forking the gentoo repo, merging your ebuilds and creating a pull request out of it, no?

The tedious work has already been done, the ebuilds work and comply to the requirements! Let's bring compiz-reloaded back into Gentoo!

Cheers,
-N

ethus3h commented:
Maybe I can do that. Thanks @mgorny!

@nkichukov: The hard part I'm talking about is generating the commits and partial ebuilds all in the right order to make it buildable after every commit, and testing it all. (Each ebuild gets its own commit.) I think there were some ebuild changes still needed, as well, which necessitates going through the whole set of 30+ commits again, correcting them with rebase, etc, which is another several hours of not-all-that-fun work. Probably it would be easy if I were better at using rebase, but this is the only thing I need it for... Since I have it working in the overlay, merging the PR wouldn't improve my quality of life any, so I tend to procrastinate.

I'll try the force pushing the branch and re-assess the situation then. (I don't have it locally since I ran out of disk space and put it somewhere I can't find it now, but I assume it's in my GitHub fork.)

It'll be at least a few days though since I'm going camping :)

Thanks!
